### PR TITLE
[nGraph] Fix bound check for reference GatherElements

### PR DESCRIPTION
**Pull request by pavel-esir**

### CODE REVIEW

File: ngraph/core/reference/include/ngraph/runtime/reference/gather_elements.hpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ namespace ngraph
                 {
                     for (int64_t i = 0; i < indices_shape[0]; i++)
                     {
-                        if (indices[i] > data_shape[0])
+                        if (indices[i] > data_shape[0] - 1)
```
Comment (lazarevevgeny) [inline ngraph/core/reference/include/ngraph/runtime/reference/gather_elements.hpp:52]: If the `data_shape[0]` is 0 then `size_t(0 - 1) = 2^63 - 1`. Change this line to:
`if (indices[i] + 1 > data_shape[0])`

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: ngraph/core/reference/include/ngraph/runtime/reference/gather_elements.hpp (hunk 1 above)
Comment (pavel-esir) [reply ngraph/core/reference/include/ngraph/runtime/reference/gather_elements.hpp:52]: Agree that the current check is not ideal. Your check is better, but the best is to do ` if (indices[i] >= data_shape[0])`. Corrected in other places. Please review.

**Review by ilyachur — APPROVED** [review]

**Comment by pavel-esir** [comment]
pre-comit finished. @lazarevevgeny please merge

**Review by lazarevevgeny — APPROVED** [review]

**Comment by lazarevevgeny** [comment]
@nshchego, do you agree with these changes?

**Review by nshchego — APPROVED** [review]

**Pull request merged**